### DISCUSSION TITLE
PID Heart function parameters

### DISCUSSION THREAD

**Discussion by peibol90** [cagnulein/qdomyos-zwift#2731] [Ideas]
Hello there.

We come from this post in the Facebook group: [https://www.facebook.com/share/p/tDafqh7QTJg5TUgt/](url)

I was told that the "pushy" behavior that I observed on the PID response was intentional to fight with boredom. Using gym trainers from Lifefitness, they performed nicely keeping you at a constant heart rate (60% or 80% for weight loss and cardio training). I like this control because this helped me relaxing with other stuff, reading or watching tv series while pedaling.

I would like to know if there is any parameter I can customize by my own to reduce the oscillatory trend. Other thing that could be done is exposing PID parameters in the advanced settings section.

I will perform a _step response ride_ in the following days to try to get my own PID tuning parameters.

**Comment by cagnulein** [comment]
Ok let's start adding a parameter removing the pushing thing, I guess it
will solve your issue
I will do in the next days

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno dom 3 nov 2024 alle 16:28 peibol90 ***@***.***> ha
scritto:

**Reply by cagnulein** [reply]
@peibol90 are you on android or ios with qz?

**Reply by peibol90** [reply]
Android 13 TKQ1.221114.001
Xiaomi HyperOS 1.0.8.0.TKCEUXM
Poco X4 Pro 5G

**Comment by cagnulein** [comment]
ok perfect i will implement it asap

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno lun 4 nov 2024 alle ore 12:37 peibol90 ***@***.***>
ha scritto:

**Comment by cagnulein** [comment]
@peibol90 done https://github.com/cagnulein/qdomyos-zwift/actions/runs/11664419319

Google is very slow in these days, reviewing the app on the store, even the beta ones. For this reason, in order to check right now if the patch that I did is right, you can use the android build in the link (check at the bottom of the page in about 1 hour). In order to test it you need first to remove the current QZ version on your phone, install the one that you downloaded.
This version is a 15 minutes trial (you can restart it as many times as you want) and I can unlock it if you want, send me an email to roberto.viola83@gmail.com mentioning this ticket and that you are on android.

**You have to disable the new setting PID Pushy in the trainprogram options**

Let me know if you have any questions

**Reply by peibol90** [reply]
[debug-Mon_Nov_4_21_31_19_2024.log](https://github.com/user-attachments/files/17624439/debug-Mon_Nov_4_21_31_19_2024.log)
[lun. nov. 4 21_38_46 2024.zip](https://github.com/user-attachments/files/17624487/lun.nov.4.21_38_46.2024.zip)
I think this has worked nicely. A bit "rough" at the beginning but It has kept me in the zone (lower side). I have to do more testing. I didn't have more time today but I wanted to give you feedback. :) You are great, man🥇

**Comment by cagnulein** [comment]
Thanks for the feedback!


Il giorno lun 4 nov 2024 alle 23:19 peibol90 ***@***.***> ha
scritto:
